iib/web/models.py

```diff
@@ -365,12 +401,35 @@ def from_json(cls, kwargs):
         if current_user.is_authenticated:
             request_kwargs['user'] = current_user
 
-        request_kwargs['type'] = RequestTypeMapping.__members__['add'].value
-
         request = cls(**request_kwargs)
         request.add_state('in_progress', 'The request was initiated')
         return request
 
+    @staticmethod
+    def _handle_request_exclusive_params(request_kwargs):
+        """Handles database operations specific to add/rm requests."""
+
+        # Assume the request is an add request
+        key = 'bundles'
+        if request_kwargs['type'] == RequestTypeMapping.__members__['rm'].value:
+            key = 'operators'
+
+        # Validate bundles/operators are correctly provided depending on the request type
+        value = request_kwargs.pop(key)
+        if (
+            not isinstance(value, list)
+            or len(value) == 0
+            or any(not item or not isinstance(item, str) for item in value)
+        ):
+            raise ValidationError(f'"{key}" should be a non-empty array of strings')
+
+        if key == 'bundles':
+            request_kwargs['bundles'] = [
+                Image.get_or_create(pull_specification=item) for item in value
+            ]
+        else:
+            request_kwargs['operators'] = [Operator.get_or_create(name=item) for item in value]
```

Comment:
The `get_or_create` method doesn't exist on the `Operator` class. I think you could probably have a base class that has a `get_or_create` method that `Architecture`, `Image`, and `Operator` inherit from. The derived class could wrap the base class `get_or_create` by passing in the column name to filter on to the base class `get_or_create`.

Reply:
I don't know how I feel about doing this though. Because there's some validation in the Image `get_or_create` method. There's a chance we add something similar in others too.